components/Resize-node.coffee

```diff
@@ -22,48 +23,107 @@ exports.getComponent = ->
     datatype: 'integer'
     description: 'New height'
     required: false
-  c.outPorts.add 'canvas',
-    datatype: 'string'
+
+  c.outPorts.add 'out',
+    datatype: 'all'
+    description: 'Resized buffer'
+  c.outPorts.add 'factor',
+    datatype: 'number'
+    description: 'Original over resized dimensions factor'
+    required: false
+  c.outPorts.add 'original',
+    datatype: 'object'
+    description: 'Original dimension'
+    required: false
+  c.outPorts.add 'resized',
+    datatype: 'object'
+    description: 'Resized dimension'
+    required: false
+  c.outPorts.add 'metadata',
+    datatype: 'object'
+    description: 'Extracted metadata while resizing'
+    required: false
   c.outPorts.add 'error',
     datatype: 'object'
     required: false
 
   noflo.helpers.WirePattern c,
-    in: ['path']
+    in: ['in']
     params: ['width', 'height']
-    out: ['canvas']
+    out: ['out', 'factor', 'original', 'resized', 'metadata']
```

Comment:
Having both original and resized dimensions are important because we can't figure out the original dimensions from factor alone. It is useful to have those as outports, but not a necessity.

It's interesting to have metadata as outport to avoid redoing EXIF extraction in other steps: we have to extract metadata anyway (to have original dimensions for factor calc) so IMHO it makes sense to output it.